### PR TITLE
refactor(cmd): add support for cmds + cleanup

### PR DESCRIPTION
**Pull request by diamante0018**

### CODE REVIEW

File: src/client/game/game.hpp [hunk 1]
```diff
@@ -50,6 +50,13 @@ namespace game
 	private:
 		size_t address_;
 	};
+
+	struct cmd_args_t
+	{
+		CmdArgs* operator->() const;
+	};
```
Comment (momo5502) [inline src/client/game/game.hpp:57]: seems like this belongs into symbols

**Review by momo5502 — CHANGES_REQUESTED** [review]
one small thing, otherwise great contribution :)

File: src/client/game/symbols.hpp [hunk 2]
```diff
@@ -84,6 +84,14 @@ namespace game
 	// Global game definitions
 	constexpr auto CMD_MAX_NESTING = 8;
 
-	// Reimplementations
+	struct cmd_args_t
+	{
+		CmdArgs* operator->() const
```
Comment (momo5502) [inline src/client/game/symbols.hpp:89]: shouldn't this be inline?

Comment (diamante0018) [reply src/client/game/symbols.hpp:89]: image

Comment (diamante0018) [reply src/client/game/symbols.hpp:89]: I can make it static inline, which would be more appropriate imo now that you make me think about it

Comment (diamante0018) [reply src/client/game/symbols.hpp:89]: Doesn't seem to make a practical difference. Up to you, if you still want it let me know.

Comment (momo5502) [reply src/client/game/symbols.hpp:89]: ay, alright :D

**Review by momo5502 — APPROVED** [review]

**Pull request merged**